### ISSUE TITLE
Multiple JOIN Statements

### ISSUE THREAD

**Issue by calebkleveter** [vapor/fluent#451]
Fluent 3 only supports a single `JOIN` statement for a query. While this covers most use cases, it means you can't, for example, do a query over a pivot model.

Ideally, I should be able to do something like this:

```swift
ModelOne.query(on: request).join(field: \ModelPivot.modelOneID).join(field: \ModelTwo.id, to: \ModelPivot.modelTwoID)...
```

**Comment by tanner0101** [comment]
@calebkleveter it should support that. What about it is not working?

**Comment by calebkleveter** [comment]
The reason this doesn't work is because the second key-path passed into the `join` method requires that the `Root` type be equal to the model that created the query:

[![enter image description here][1]][1]

Which means I _can't_ pass in `\ModelPivot.modelTwoID` as I did in the example in the original post.

  [1]: https://i.stack.imgur.com/tYulh.png

**Comment by tanner0101** [comment]
@calebkleveter you need to do two separate joins for that to work. One from the local table to the pivot. Then one from the pivot to the foreign table.

**Comment by calebkleveter** [comment]
Do you have an example? I thought I was try to do two separate joins.

**Comment by tanner0101** [comment]
Ah, I see the problem now. Fluent needs to allow something like this:
```swift
let users = try User.query(on: conn)
    .join(PetUser.self, field: \.userID, to: \.id)
    .join(Pet.self, field: \.id, to: PetUser.self, field: \.id)
```

**Comment by tanner0101** [comment]
This is fixed in #482, thanks!
